src/main/dumpers/schemas/metadatalist.h

```diff
@@ -0,0 +1,98 @@
+/**
+ * =============================================================================
+ * DumpSource2
+ * Copyright (C) 2024 ValveResourceFormat Contributors
+ * 
+ * Source2Gen
+ * Copyright (C) 2024 neverlosecc
+ * =============================================================================
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License, version 3.0, as published by the
+ * Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+#include <array>
+#include "utils/hash.h"
+
+// List of printable metadata entries that are not tied to any structures.
+// These are used to determine if we can include the corresponding metadata entry value in the dump.
+// Original list sourced from Source2Gen project.
+
+constinit std::array string_metadata_entries = {
```

Comment:
I believe this is way too overengineered, we do not care about few ms of performance, but do care about the code readibility. I am for using std map with std strings that will get hashed in std lib itself and map it to an enum of the different type values. Then resolve those enums in a switch case to the proper string value.